### PR TITLE
query and update animals table

### PR DESCRIPTION
**Pull request by mosidrum**
query and update animals table

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,3 +4,10 @@ INSERT INTO animals VALUES (1, 'Agumon', '2020-03-02',0,true,10.23)
 INSERT INTO animals VALUES (2, 'Gabumon', '2018-11-15', 2, true, 8 )
 INSERT INTO animals VALUES (3, 'Pikachu', '2021-01-07', 1, false, 15.04)
 INSERT INTO animals VALUES (4, ' Devimon', '2017-05-17', 5, true, 11)
+INSERT INTO animals VALUES (5, ' Charmander', '2020-02-08', 0, false, -11)
+INSERT INTO animals VALUES(6, 'Plantmon', '2021-11-15', 2, true, -5.7),
+(7, 'Squirtle', '1993-04-02', 3, false,  -12.13),
+(8, 'Angemon', '2005-06-12', 1, true, -45),
+(9, ' Boarmon', '2007-06-07', 7, true, 20.4),
```
Comment (muneebulrehman) [inline data.sql]: You have made a typo with the date_of_birth of Boarmon. It is 2005/06/07 and not 2007/06/07. Kindly fix this mistake.

File: schema.sql [hunk 2]
```diff
@@ -8,3 +8,4 @@ CREATE TABLE animals (
 	neutered BOOLEAN,
 	weight_kg DECIMAL (4,2)
 )
+ALTER TABLE animals ADD COLUMN species;
```
Comment (muneebulrehman) [inline schema.sql:11]: Kindly make the species varchar as per the project requirements. ⬇️
image

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,58 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
 SELECT (name, escape_attempts) FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3 ;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3 ;
+ALTER TABLE animals
+ADD COLUMN species varchar(255);
```
Comment (muneebulrehman) [inline queries.sql]: Any code that deals with creating/altering the database should go into the schema.sql file. Kindly remove his line of code from here.

File: queries.sql [hunk 4]
```diff
@@ -7,4 +7,58 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+BEGIN
+UPDATE animals
+SET species = 'unspecifies';
```
Comment (muneebulrehman) [inline queries.sql]: As per the project requirements, once you have set the species = 'unspecified', you need to verify whether the changes were made by selecting all the species from animals. Kindly update your code accordingly

File: queries.sql [hunk 5]
```diff
@@ -7,4 +7,58 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+ROLLBACK --species column is back to null
```
Comment (muneebulrehman) [inline queries.sql:16]: As per the project requirements, after ROLLBACK you again need to verify whether the ROLLBACK was successful by selecting all the animals in the table. Kindly update your code accordingly.

File: queries.sql [hunk 6]
```diff
@@ -7,4 +7,58 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+--What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+--What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
```
Comment (muneebulrehman) [inline queries.sql:65]: Screenshots for both of these queries are not included in the PR description. Kindly include the screenshots for these queries before submitting for a re-review.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @mosidrum , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- All queries included. 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @mosidrum, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**